src/sql/base/browser/ui/table/table.ts

```diff
@@ -405,4 +463,59 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 	public get container(): HTMLElement {
 		return this._tableContainer;
 	}
+
+	// Checks if the current table is a tree grid. Every tree grid should have a column with expandableColumnFormatter
+	private isTreeGrid(): boolean {
+		return this._grid.getColumns().filter(c => c.formatter === treeGridExpandableColumnFormatter).length > 0;
```

Comment:
looks like you should introduce a property in the column definition, this doesn't seem to be a proper way to achieve what you want.

Reply:
This is not needed as I have made a new TreeGrid component.